turtle_tf2_cpp/src/turtle_tf2_listener.cpp

```diff
@@ -88,42 +66,92 @@ class FrameListener : public rclcpp::Node
     std::string fromFrameRel = target_frame_.c_str();
     std::string toFrameRel = "turtle2";
 
-    geometry_msgs::msg::TransformStamped transformStamped;
-
-    // Look up for the transformation between target_frame and turtle2 frames
-    // and send velocity commands for turtle2 to reach target_frame
-    try {
-      transformStamped = tf_buffer_->lookupTransform(
-        toFrameRel, fromFrameRel,
-        tf2::TimePoint(),
-        500ms);
-    } catch (tf2::LookupException & ex) {
-      RCLCPP_INFO(this->get_logger(), "transform not ready");
-      return;
+    if (!turtle_spawned_) {
+      // Check if the service is available
+      while (!spawner_->wait_for_service(1s)) {
+        if (!rclcpp::ok()) {
+          RCLCPP_ERROR(
+            this->get_logger(),
+            "Interrupted while waiting for the service. Exiting."
+          );
+          timer_->cancel();
+          return;
+        }
+        RCLCPP_INFO(
+          this->get_logger(),
+          "Service not available, waiting again..."
+        );
+      }
+
+      // Initialize request with turtle name and coordinates
+      // Note that x, y and theta are defined as floats in turtlesim/srv/Spawn
+      auto request = std::make_shared<turtlesim::srv::Spawn::Request>();
+      request->x = 4.0;
+      request->y = 2.0;
+      request->theta = 0.0;
+      request->name = "turtle2";
+      // Call request
+      auto result = spawner_->async_send_request(request);
+
+      turtle_spawned_ = true;
```

Comment:
Before setting `turtle_spawned_` to true, I feel like we should check whether the service request was successful.  That would require storing the result, and then on subsequent `on_timer` callbacks checking whether it completed successfully.

Reply:
Partially solved in https://github.com/ros/geometry_tutorials/pull/53/commits/4a380e7cc5b293228d224a429d4d5fbce7a168db. I got the idea to acquire the result and wait for the next timer call to fetch the result, but I have not made any substantial checks there. Do you know how could I improve that part?